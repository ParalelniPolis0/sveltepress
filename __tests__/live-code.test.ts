import { describe, expect, it } from 'vitest'
import { compile } from 'mdsvex'
import liveCode from '../src/markdown/live-code'

const md = `
### title level 3
<script context="module">
  const a = 'a'
  export const b = 'b'
</script>
<script>
  import Foo from '/path/to/Foo.svelte'
</script>

<Foo />

\`\`\`svelte live
<script>
  let count = 1
</script>

<button on:click={() => count++}>
  You've clicked {count} times
</button>
\`\`\`
`

describe('live code', () => {
  it('simple parse', async () => {
    const { code } = await compile(md, {
      remarkPlugins: [liveCode],
    }) || { code: '' }

    expect(code).toMatchSnapshot()
  })
})
