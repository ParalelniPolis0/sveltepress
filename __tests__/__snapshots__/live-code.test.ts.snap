// Vitest Snapshot v1

exports[`live code > simple parse 1`] = `
"<script context=\\"module\\">
import CExpansion from '@sveltepress/theme-default/CExpansion.svelte'
import Link from '@sveltepress/theme-default/Link.svelte'
import CopyCode from '@sveltepress/theme-default/CopyCode.svelte'
import Tabs from '@sveltepress/theme-default/Tabs.svelte'
import TabPanel from '@sveltepress/theme-default/TabPanel.svelte'
import LiveCodea0d9514c47e from '$sveltepress/live-code/LiveCodea0d9514c47e.svelte'
  const a = 'a'
  export const b = 'b'
</script>
<script>
  import Foo from '/path/to/Foo.svelte'
</script>

<h3>title level 3</h3>
<Foo />
<div class=\\"svp-live-code--container\\"><div class=\\"svp-live-code--demo\\"><LiveCodea0d9514c47e /></div><CExpansion title=\\"Click fold/expand code\\" reverse={true}><pre class=\\"language-svelte\\">{@html \`<code class=\\"language-svelte\\"><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>script</span><span class=\\"token punctuation\\">></span></span><span class=\\"token script\\"><span class=\\"token language-javascript\\">
  <span class=\\"token keyword\\">let</span> count <span class=\\"token operator\\">=</span> <span class=\\"token number\\">1</span>
</span></span><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>script</span><span class=\\"token punctuation\\">></span></span>

<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>button</span> <span class=\\"token attr-name\\"><span class=\\"token namespace\\">on:</span>click=</span><span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">&#123;</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token operator\\">=></span> count<span class=\\"token operator\\">++</span><span class=\\"token punctuation\\">&#125;</span></span><span class=\\"token punctuation\\">></span></span>
  You've clicked <span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">&#123;</span>count<span class=\\"token punctuation\\">&#125;</span></span> times
<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>button</span><span class=\\"token punctuation\\">></span></span></code>\`}</pre></CExpansion></div>
"
`;
